File: ruby/lib/Sikaru/run_schedules/types/schedule_input.rb
# frozen_string_literal: true

module Sikaru
  module RunSchedules
    module Types
      class ScheduleInput < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :cron, -> { String }, optional: true, nullable: false

        field :idempotency_key, -> { String }, optional: true, nullable: false

        field :input, -> { Internal::Types::Hash[String, Object] }, optional: false, nullable: false

        field :interval_seconds, -> { Integer }, optional: true, nullable: false

        field :session_id, -> { String }, optional: false, nullable: false

        field :session_mode, -> { Sikaru::RunSchedules::Types::ScheduleInputSessionMode }, optional: true, nullable: false

        field :timezone, -> { String }, optional: true, nullable: false
      end
    end
  end
end

File: typescript/src/api/resources/runSchedules/client/requests/ScheduleInput.ts

/**
 * @example
 *     {
 *         input: {
 *             "key": "value"
 *         },
 *         interval_seconds: 1,
 *         session_id: "session_id"
 *     }
 */
export interface ScheduleInput {
    idempotency_key?: string | null;
    input: Record<string, unknown>;
    interval_seconds: number;
    session_id: string;
}
